src/useRouteQuery/useRouteQuery.ts

```diff
@@ -0,0 +1,112 @@
+import { reactive, watch } from 'vue'
+import { LocationQuery, LocationQueryValue, useRoute, useRouter } from 'vue-router'
+
+type UseRouteQuery = {
+  query: LocationQuery,
+  clear: () => void,
+  set: (key: string, value: string) => void,
+  get: (key: string) => LocationQueryValue | LocationQueryValue[],
+  remove: (key: string) => void,
+}
+
+type QueryOperation = (query: LocationQuery) => LocationQuery
+
+export function useRouteQuery(): UseRouteQuery {
+  const route = useRoute()
+  const router = useRouter()
+  const operations = reactive<QueryOperation[]>([])
+  let interval: ReturnType<typeof setTimeout>
+
+  const clear = (): void => {
+    router.push({ query: {} })
```

Comment:
are you worried at all about conflicts between 2 components that both want to use this composition? I wonder if it would be worth the complexity to have `clear` only remove values that were explicitly added by this instance. I don't feel strongly, just thinking out loud

Reply:
No not really. This composition is designed to give you the whole query. I think it would be more confusing if it tried to scope the query just to specific properties.